### ISSUE TITLE
Android Autofill not working on latest stable (2.10.3)

### ISSUE THREAD

**Issue by aya2453** [flutter/flutter#100618]
Autofill is not working when using 2.10.3 (also didn't work for 2.10.1) especially when I use `AutofillHints.creditCardNumber` `AutofillHints.address` and so on. `AutofillHints.name` works fine. This problem was also reported in another issue, but it's still not solved although the issue was closed and release notes also mentions it as resolved in 2.10.1.
older issue: https://github.com/flutter/flutter/issues/94043 / https://github.com/flutter/flutter/issues/98112

**Comment by HansMuller** [comment]
CC @LongCatIsLooong

**Comment by LongCatIsLooong** [comment]
the old issue was fixed and cherrypicked to 2.10.3. I can confirm "email" works fine on 2.10.3 Was "creditCardNumber" or "address" working at some point? Depending on the autofill service you're using, you may need to set those up in the system "autofill service" settings and provide relevant information to the autofill service for those to work.

**Comment by maheshmnj** [comment]
Hi @aya2453, Thanks for filing the issue. I can confirm that name, email, Password works. on `stable 2.10.3`.
But `postalAddress` (since `AutofillHints.address` does not exist)  and `creditCardNumber` doesn't work (I have saved these details, but the autoFill doesn't show up).

The issue for `creditCardNumber` is being tracked here https://github.com/flutter/flutter/issues/98762 please follow up there for further Updates closing this issue as a duplicate. In case you disagree feel free to write in the comments and we will reopen it.



https://user-images.githubusercontent.com/31410839/159864161-174200dc-a9b9-4efa-b1e6-b9c55f084e16.mov



<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  List<String> hints = [
    AutofillHints.name,
    AutofillHints.email,
    AutofillHints.password,
    AutofillHints.creditCardNumber,
    AutofillHints.addressCity,
    AutofillHints.addressState,
    AutofillHints.addressCityAndState,
    AutofillHints.postalAddress,
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: AutofillGroup(
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                SizedBox(
                  height: kToolbarHeight,
                ),
                for (int i = 0; i < hints.length; i++)
                  Column(
                    children: [
                      TextField(
                        decoration: InputDecoration(hintText: hints[i]),
                        obscureText: hints[i] == AutofillHints.password,
                        keyboardType: hints[i] == AutofillHints.creditCardNumber
                            ? TextInputType.number
                            : hints[i] == AutofillHints.email
                                ? TextInputType.emailAddress
                                : TextInputType.text,
                        autofillHints: [hints[i]],
                      ),
                      const SizedBox(
                        height: 16,
                      ),
                    ],
                  ),
                ElevatedButton(
                  child: const Text("Save"),
                  onPressed: () {
                    TextInput.finishAutofillContext(shouldSave: true);
                    debugPrint("Saving");
                  },
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

```
</details>  

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.3 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (20 hours ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.5:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.103, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
    • Flutter version 2.13.0-0.0.pre.103 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 509ddfda5c (7 hours ago), 2022-03-15 21:10:18 -0700
    • Engine revision dc22c4c7fe
    • Dart version 2.17.0 (build 2.17.0-202.0.dev)
    • DevTools version 2.11.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that
        responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.2.1 21D62 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>
